gateway/modules/eventing/routine.go

```diff
@@ -12,28 +14,137 @@ import (
 
 func (m *Module) createProcessUpdateEventsRoutine() {
 	m.updateEventC = make(chan *queueUpdateEvent, 1000)
-	for i := 0; i < 50; i++ {
+	for i := 0; i < 25; i++ {
 		go m.routineProcessUpdateEvents()
 	}
 }
 
 func (m *Module) routineProcessUpdateEvents() {
-	for ev := range m.updateEventC {
-		m.queueUpdateEvent(ev)
+	for {
+		select {
+		case <-m.globalCloserChannel:
+			fmt.Println("Closing routineProcessStaged")
+			return
+		case ev := <-m.updateEventC:
+			m.queueUpdateEvent(ev)
+			m.deleteEventFromProcessingEventsMapChannel <- []string{ev.req.Find["_id"].(string)}
+		}
+	}
+}
+
+func (m *Module) routineUpdateEventsStatusInDB(updateChan chan *queueUpdateEvent) {
+	duration := 1 * time.Second
+	t := time.NewTimer(duration)
+	defer t.Stop()
+	arr := make([]*queueUpdateEvent, 0)
+	for {
+		select {
+		case <-m.globalCloserChannel:
+			close(m.updateFailedEventInDBChannel)
+			fmt.Println("Closing routineUpdateEventsStatusInDB")
+			return
+		case ev := <-updateChan:
+			arr = append(arr, ev)
+		case <-t.C:
+			if len(arr) == 0 {
+				t.Reset(duration)
+				continue
+			}
+			eventIDs, updateRequest := m.generateInOperatorUpdateRequest(arr)
+			m.queueUpdateEvent(updateRequest)
+			m.deleteEventFromProcessingEventsMapChannel <- eventIDs
+			arr = make([]*queueUpdateEvent, 0)
+			t.Reset(duration)
+		}
+	}
+}
+
+func (m *Module) routineDeleteEventsFromSyncMap() {
+	duration := 5 * time.Second
+	t := time.NewTimer(duration)
+	defer t.Stop()
+
+	activeArr := make([]string, 0)
+	for {
+		select {
+		case <-m.globalCloserChannel:
+			fmt.Println("Closing routineDeleteEventsFromSyncMap")
+			return
+		case eventIDs := <-m.deleteEventFromProcessingEventsMapChannel:
+			activeArr = append(activeArr, eventIDs...)
+		case <-t.C:
+			passiveArr := activeArr
```

Comment:
This is wrong. Passive array needs to be declared alongside active array. Set the value of the passive array to active array after you done deleting values of previouspassive array. Then make a new active array